### PR TITLE
New vital point class

### PR DESCRIPTION
**Pull request by aaron-bray**

### CODE REVIEW

File: vital/types/point.h [hunk 1]
```diff
@@ -52,25 +52,26 @@ template < unsigned N, typename T >
...
 
   virtual ~point() = default;
 
   vector_type value() const { return m_value; }
   void set_value(vector_type v) { m_value = v; }
   
-  covariance_<N, float> covariance() const { return m_covariance; }
-  void set_covariance(covariance_<N, float> v)
+  covariance_type covariance() const { return m_covariance; }
+  void set_covariance(covariance_type v)
```
[the remark this answers is not in the document]
Comment (aaron-bray) [reply vital/types/point.h]: you ok with no ref?
If so, we can merge it in.

**Comment by aaron-bray** [comment]
@mwoehlke-kitware  I am going to merge this today, unless you want change set_covariance to use a &

File: vital/types/point.cxx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+template class point< 2, int >;
+template class point< 2, double >;
+template class point< 2, float >;
+template class point< 3, double >;
+template class point< 3, float >;
+template class point< 4, double >;
+template class point< 4, float >;
+
+template < unsigned N, typename T >
```
[the remark this answers is not in the document]
Comment (mwoehlke-kitware) [reply vital/types/point.cxx]: So... I seem to recall it being harder than this, but... see e.g. https://godbolt.org/z/1uXDmE or https://gist.github.com/mwoehlke-kitware/45f67bbdd0495f3845282af01964834f. ISTR that there was some obnoxious manner in which one needed to add `extern` to the instantiation (i.e. depended on *both* whether or not you are on Windows, and whether you are importing or exporting), but cursory testing seems to indicate that it should work without that. However, I haven't tried it in a real project.

[the remark this answers is not in the document]
Comment (aaron-bray) [reply vital/types/point.cxx]: I have used this kind of forward declaration before in projects and seems to work fine. 
I always thought this was to catch any problems with instantiating templates of know types at compile time of the library, so when users use these known template types, it won't barf on them, but on the template developer...

**Pull request merged**